### PR TITLE
Chance card movement bug fixes #16

### LINKED ISSUE #16: update movement when chance card drawn
You'll notice when a player draws a chance / cc chest card and it moves their token (for example, to reading railroad), the player will not be charged if another player owns it nor will they have the option to buy it. Adding a bit more logic to the respective *match / case* line here should help. 

The offending chance cards are as follows (see ascii.txt):  5, 6, 7, 10 (i think), 14

Please test each before you submit a PR.

### PR DESCRIPTION
**Pull request by RunDMC2**
Fixed movement bug when landing on a chance or community chest space where player can't buy a space they land on when moved by a card

Fixed minor printing bug where space 10 and 7 printed at the same location

Fixed modularizing bug where cards.py needs but doesn't have access to update_history()

### CODE REVIEW

File: .idea/.gitignore
Comment (adamgulde) [inline .idea/.gitignore:1]: please do not commit the .idea folder from your IDE

File: board.py
Comment (adamgulde) [inline board.py:1]: nice catch!

**Review by adamgulde — CHANGES_REQUESTED** [review]
this seems to work from some testing I did, but at the moment will create conflicts because you did not push with the most recently updated code from main ufosc branch. Please see the new steps 3-6 for submitting changes from CONTRIBUTING.md. Following these guidelines will make review process much faster.

**Comment by RunDMC2** [comment]
just got rid of the .idea folder, and as for the code from the ufosc branch everything on my end says my fork is synced and my branch is up to date when I commit the changes I've made, let me know if any further changes need to be made

**Comment by adamgulde** [comment]
> just got rid of the .idea folder, and as for the code from the ufosc branch everything on my end says my fork is synced and my branch is up to date when I commit the changes I've made, let me know if any further changes need to be made

ok weird- there's some stuff coming up when i look at the pr. can you come to cc today to look over this w me or send me your discord name?

**Comment by RunDMC2** [comment]
Yeah sounds good I'll be at cc today we can look over it

File: monopoly.py [hunk 1]
```diff
@@ -399,11 +399,13 @@ def player_roll(num_rolls):
                 update_history(f"Player {turn} rolled doubles three times\n in a row!")
                 update_history(f"Player {turn} is going to jail!")
                 players[turn].jail = True
-                board.update_location(players[turn], -1, update_history)
+                board.update_location(players[turn], -1)
         refresh_board()
         #if player rolled their third double they will be in jail and their location doesn't update
         if players[turn].jail == False:
-            board.update_location(players[turn], dice[0] + dice[1], update_history)
+            if (players[turn].location + dice[0] + dice[1]) > 39:  # checks if player passed go
```
Comment (RunDMC2) [inline monopoly.py:406]: checks if player passes go only after they roll the dice

File: monopoly.py [hunk 2]
```diff
@@ -417,11 +419,19 @@ def player_roll(num_rolls):
...
                         update_history(f"{players[turn].name} drew a Community Chest card! {card}")
+                        if old_loc > new_loc and new_loc != 10 and new_loc != players[turn].location - 3:  #check if chance card made player pass go
+                            update_history(f"Player {players[turn].order} passed Go and received $200")
                     case -4: #chance
-                        card = decks.draw_chance(players[turn], board, players, update_history)
+                        old_loc = players[turn].location
+                        card = decks.draw_chance(players[turn], board, players)
+                        new_loc = players[turn].location
                         update_history(f"{players[turn].name} drew a Chance card! {card}")
+                        if old_loc > new_loc and new_loc != 10 and new_loc != players[turn].location - 3:  #check if chance card made player pass go
```
Comment (RunDMC2) [inline monopoly.py:433]: stores location before and after drawing a chance/community chest card, then compares the old and new locations to check if player has passed go

**Review by adamgulde — APPROVED** [review]
lgtm! great work :)

**Pull request merged**